src/js/helpers.test.js

```diff
@@ -164,7 +164,7 @@ wget -O- https://get.mender.io | sudo bash -s -- --demo --commercial --jwt-token
       expect(code).toMatch(
         `JWT_TOKEN="omnomnom"
 TENANT_TOKEN="token"
-wget -O- https://get.mender.io/staging | sudo bash -s -- --demo -c experimental --commercial --jwt-token $JWT_TOKEN -- --quiet --device-type "raspberrypi3" --tenant-token $TENANT_TOKEN --demo --hosted-mender`
+wget -O- https://get.mender.io/staging | sudo bash -s -- --demo -c experimental --commercial --jwt-token $JWT_TOKEN -- --quiet --device-type "raspberrypi3" --tenant-token $TENANT_TOKEN --demo --server-url https://staging.hosted.mender.io --server-cert=""`
```

Comment:
afaik one of the `--demo` flags might trust the certificate from the url otherwise (but we can assume the cert is valid & trusted in production installations) - but I guess @lluiscampos knows best!

Reply:
Wait a second! I am partially wrong. `--server-url` is a requirement for production installations (for installations when the certificated is not CA distributed), not related to the "demo server" settings.

So we can do what I suggested about `--demo-polling` (for correctness) but we still need to keep `server-cert`:
```suggestion
wget -O- https://get.mender.io/staging | sudo bash -s -- --demo -c experimental --commercial --jwt-token $JWT_TOKEN -- --quiet --device-type "raspberrypi3" --tenant-token $TENANT_TOKEN --demo-polling --server-url https://staging.hosted.mender.io --server-cert=""`
```

Reply:
Based on your explanation I guess you meant `--server-cert` as a requirement (since `--server-url` is provided)? Last year there was [this](https://github.com/mendersoftware/gui/commit/70224fa97df0c5cfed2b24bcfef876db1f014892) decision to expect a CA trusted certificate to be present in prod installs... so should we then no longer expect this?
Also based on your explanation I would like to stick to `--demo` for brevity (and since we will set the `--server-url` in any case)...

And just to clarify in general: What exactly does the setup change in the certificate handling if it is called with `--server-cert=""`? and what if we provide a cert?

Reply:
By looking at the code, `--server-cert` is required in all cases except 1) `--hosted` is provided (because we assume HM is CA trusted) or 2) `--demo[-server]` is provided (because we know where the self-signed certificate is located in the filesystem).

Let's look at it from a different point of view: `mender setup` is by design an interactive command. It is meant for user to input things. So basically not setting `--server-cert=""` will make the command prompt the user where the server cert is (which could by blank, in which case means CA backed).

> Also based on your explanation I would like to stick to `--demo` for brevity (and since we will set the `--server-url` in any case)...

As you wish :+1: It is a deprecated flag but we don't have plans to remove it any time soon. We probably remove it on the C++ client.

> And just to clarify in general: What exactly does the setup change in the certificate handling if it is called with `--server-cert=""`? and what if we provide a cert?

If blank, nothing happens. If provided, it is set as [`ServerCertificate` configuration field](https://docs.mender.io/client-installation/configuration-file/configuration-options#servercertificate).

Reply:
Thank you very much for the explanation 👌 Then I'll merge as it is now!

> We probably remove it on the C++ client.

On the topic of changes coming with the re-write: Would it be an option to change the setup call to only be interactive if no options are supplied at all (and assume defaults for any options not supplied)?
It might be just the case in my "tool bubble", but there this approach is very common...

Reply:
> Would it be an option to change the setup call to only be interactive if no options are supplied at all (and assume defaults for any options not supplied)? It might be just the case in my "tool bubble", but there this approach is very common...

Thanks for the suggestion.

I actually thought about exactly this long time ago, but then I was afraid of breaking backwards compatibility. However the C++ client will be launched as a major version, so it is a great opportunity to introduce this change. (Which, I agree with you, is much more user friendly: either full interactive mode or only set the configuration fields passed by cli options).

I am making a note in the planning document and I'll bring it up later down the road with the client team.